pkg/liqo-controller-manager/resource-request-controller/external-monitor.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcerequestoperator
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// ExternalResourceMonitor is an object that keeps track of the cluster's resources.
+type ExternalResourceMonitor struct {
+	BrokerClient
+}
+
+// NewExternalMonitor creates a new ExternalResourceMonitor.
+func NewExternalMonitor(address string) (*ExternalResourceMonitor, error) {
+	conn, err := grpc.Dial(address, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		klog.Errorf("Could not connect to external resource monitor at %s: %s", address, err)
+		return nil, err
+	}
+	client := NewBrokerClient(conn)
+	return &ExternalResourceMonitor{
+		BrokerClient: client,
+	}, nil
+}
+
+// Register sets an update notifier.
+func (m *ExternalResourceMonitor) Register(notifier ResourceUpdateNotifier) {
+	stream, err := m.BrokerClient.Subscribe(context.Background(), &SubscribeRequest{})
```

Comment:
```suggestion
func (m *ExternalResourceMonitor) Register(ctx context.Context, notifier ResourceUpdateNotifier) {
	stream, err := m.BrokerClient.Subscribe(ctx, &SubscribeRequest{})
```

the same for the other functions